src/sentry/south_migrations/0468_recent_search.py

```diff
@@ -0,0 +1,1362 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import SchemaMigration
+
+
+class Migration(SchemaMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = False
```

Comment:
@mattrobenolt Not sure whether this migration is dangerous, since it adds fk relations to both organization and user.

Reply:
Yeah, so I'm not sure. So let's flip it to True, and run it by hand. :)

Can you dump the SQL equivalent to this migration for me?

Reply:
Here's the migration:

```
CREATE TABLE "sentry_recentsearch" (
  "id" bigserial NOT NULL PRIMARY KEY,
  "organization_id" bigint NOT NULL,
  "user_id" integer NOT NULL,
  "type" smallint CHECK ("type" >= 0) NOT NULL,
  "query" text NOT NULL,
  "last_seen" timestamp with time zone NOT NULL,
  "date_added" timestamp with time zone NOT NULL
);
ALTER TABLE "sentry_recentsearch" ADD CONSTRAINT "sentry_recentsearch_user_id_61eff117f312dea5_uniq" UNIQUE ("user_id", "organization_id", "type", "query");
ALTER TABLE "sentry_recentsearch" ADD CONSTRAINT "organization_id_refs_id_7344d24c" FOREIGN KEY ("organization_id") REFERENCES "sentry_organization" ("id") DEFERRABLE INITIALLY DEFERRED;
CREATE INDEX "sentry_recentsearch_organization_id" ON "sentry_recentsearch" ("organization_id");
ALTER TABLE "sentry_recentsearch" ADD CONSTRAINT "user_id_refs_id_e212afed" FOREIGN KEY ("user_id") REFERENCES "auth_user" ("id") DEFERRABLE INITIALLY DEFERRED;
```

Reply:
@mattrobenolt To make mysql work I've had to change the django column type for `query` to `models.CharField(max_length=1000)`. I'm pretty sure we can still back this with a `text` column in Postgres right? Or I can just change the field to be a `varchar(1000)`

Reply:
lol yup, welcome to MySQL.

Yeah, we can use CharField, but use a `text` column in the migration since that works just fine in Postgres.

It's also possible to consider using a `query_hash` value, and unique on that instead. So like, md5 hash the query, then apply the index on that. This has value in not indexing an infinitely large field too.

Reply:
Yeah, that's not a bad idea. I wouldn't expect queries to get too long in most cases, but someone will make a giant search at some point for sure.

Reply:
@mattrobenolt Here's the updated ddl:

```
CREATE TABLE "sentry_recentsearch" (
  "id" bigserial NOT NULL PRIMARY KEY,
  "organization_id" bigint NOT NULL,
  "user_id" integer NOT NULL,
  "type" smallint CHECK ("type" >= 0) NOT NULL,
  "query" text NOT NULL,
  "query_hash" varchar(32) NOT NULL,
  "last_seen" timestamp with time zone NOT NULL,
  "date_added" timestamp with time zone NOT NULL
);
ALTER TABLE "sentry_recentsearch" ADD CONSTRAINT "sentry_recentsearch_user_id_6a0df1c80b29d349_uniq" UNIQUE ("user_id", "organization_id", "type", "query_hash");
ALTER TABLE "sentry_recentsearch" ADD CONSTRAINT "organization_id_refs_id_7344d24c" FOREIGN KEY ("organization_id") REFERENCES "sentry_organization" ("id") DEFERRABLE INITIALLY DEFERRED;
CREATE INDEX "sentry_recentsearch_organization_id" ON "sentry_recentsearch" ("organization_id");
ALTER TABLE "sentry_recentsearch" ADD CONSTRAINT "user_id_refs_id_e212afed" FOREIGN KEY ("user_id") REFERENCES "auth_user" ("id") DEFERRABLE INITIALLY DEFERRED;
```